control/controlknobs/controlknobs.go

```diff
@@ -76,6 +76,10 @@ type Knobs struct {
 	// AppCStoreRoutes is whether the node should store RouteInfo to StateStore
 	// if it's an app connector.
 	AppCStoreRoutes atomic.Bool
+
+	// UserDialUseRoutes is whether tsdial.Dialer.UserDial should use routes to determine
```

Comment:
This nodeAttr does two kinda separate but related things now, right? UserDial behavior change vs UserDial-for-DNS change?

Maybe that should be two nodeAttrs.

Reply:
I think we shouldn't use UserDial for DNS unless `UserDialUseRoutes` has been enabled and `UserDial` works as `SystemDial` when connecting to public DNS servers.

We _could_ make two `nodeAttrs` and check both in the forwarder. But I'm a little worried that it would be too confusing.

Reply:
At least update the comment here and in tailcfg.go to say the two separate-but-related things it does, then.

Reply:
Done.